openmetadata-service/src/main/java/org/openmetadata/service/apps/bundles/searchIndex/SearchIndexApp.java

```diff
@@ -239,174 +268,252 @@ public void updateRecordToDb(JobExecutionContext jobExecutionContext) {
-                try {
-                  reCreateIndexes(entityType);
-                  contextData.put(ENTITY_TYPE_KEY, entityType);
-
-                  while (!paginatedSource.isDone()) {
-                    Object resultList = paginatedSource.readNext(null);
-                    if (resultList == null) {
-                      break;
-                    }
-
-                    if (!TIME_SERIES_ENTITIES.contains(entityType)) {
-                      List<String> entityNames =
-                          getEntityNameFromEntity(
-                              (ResultList<? extends EntityInterface>) resultList, entityType);
-                      contextData.put(ENTITY_NAME_LIST_KEY, entityNames);
-                      processEntity(
-                          (ResultList<? extends EntityInterface>) resultList,
-                          contextData,
-                          paginatedSource);
-                    } else {
-                      List<String> entityNames =
-                          getEntityNameFromEntityTimeSeries(
-                              (ResultList<? extends EntityTimeSeriesInterface>) resultList,
-                              entityType);
-                      contextData.put(ENTITY_NAME_LIST_KEY, entityNames);
-                      processEntityTimeSeries(
-                          (ResultList<? extends EntityTimeSeriesInterface>) resultList,
-                          contextData,
-                          paginatedSource);
-                    }
-                    synchronized (jobDataLock) {
-                      updateStats(entityType, paginatedSource.getStats());
-                    }
-                    sendUpdates(jobExecutionContext);
-                  }
-
-                } catch (SearchIndexException e) {
-                  synchronized (jobDataLock) {
-                    jobData.setStatus(EventPublisherJob.Status.RUNNING);
-                    jobData.setFailure(e.getIndexingError());
-                    paginatedSource.updateStats(
-                        e.getIndexingError().getSuccessCount(),
-                        e.getIndexingError().getFailedCount());
-                    updateStats(entityType, paginatedSource.getStats());
-                  }
-                  sendUpdates(jobExecutionContext);
-                } catch (Exception e) {
-                  synchronized (jobDataLock) {
-                    jobData.setStatus(EventPublisherJob.Status.FAILED);
-                    jobData.setFailure(
-                        new IndexingError()
-                            .withErrorSource(IndexingError.ErrorSource.JOB)
-                            .withMessage(e.getMessage()));
-                  }
-                  sendUpdates(jobExecutionContext);
-                  LOG.error("Unexpected error during reindexing for entity {}", entityType, e);
-                }
-              });
-
-      futures.add(future);
-    }
+    int numProducers = jobData.getEntities().size();
```

Comment:
@harshach number of producers can be 39 , if the user select all entities might not be a good idea to select this equal to entities count

Reply:
what would be the issue with that?

Reply:
for moderate operations it should be ok , but in cases where there are long running process which can be in our cases fixed thread pool doesn't scale well , and newCachedThreadPool is more suited